examples/binary_segmentation_buildings.py

```diff
@@ -0,0 +1,532 @@
+# ----------------------------
+class Dataset(BaseDataset):
+    """
+    A custom dataset class for binary segmentation tasks.
+
+    Parameters:
+    ----------
+
+    - images_dir (str): Directory containing the input images.
+    - masks_dir (str): Directory containing the corresponding masks.
+    - INPUT_IMAGE_RESHAPE (tuple, optional): Desired shape for the input
+      images and masks. Default is (320, 320).
+    - FOREGROUND_CLASS (int, optional): The class value in the mask to be
+      considered as the foreground. Default is 1.
+    - augmentation (callable, optional): A function/transform to apply to the
+      images and masks for data augmentation.
+    """
+
+    def __init__(
+        self,
+        images_dir,
+        masks_dir,
+        INPUT_IMAGE_RESHAPE=(320, 320),
+        FOREGROUND_CLASS=1,
+        augmentation=None,
+    ):
+        self.ids = os.listdir(images_dir)
+        self.images_fps = [os.path.join(images_dir, image_id) for image_id in self.ids]
+        self.masks_fps = [os.path.join(masks_dir, image_id) for image_id in self.ids]
+
+        self.INPUT_IMAGE_RESHAPE = INPUT_IMAGE_RESHAPE
+        self.FOREGROUND_CLASS = FOREGROUND_CLASS
+        self.augmentation = augmentation
+
+    def __getitem__(self, i):
+        """
+        Retrieves the image and corresponding mask at index `i`.
+
+        Parameters:
+        ----------
+
+        - i (int): Index of the image and mask to retrieve.
+        Returns:
+        - A tuple containing:
+            - image (torch.Tensor): The preprocessed image tensor of shape
+            (1, INPUT_IMAGE_RESHAPE) - e.g., (1, 320, 320) - normalized to [0, 1].
+            - mask_remap (torch.Tensor): The preprocessed mask tensor of
+            shape INPUT_IMAGE_RESHAPE with values 0 or 1.
+        """
+        # Read the image
+        image = cv2.imread(
+            self.images_fps[i], cv2.IMREAD_GRAYSCALE
+        )  # Read image as grayscale
+        image = np.expand_dims(image, axis=-1)  # Add channel dimension
+
+        # resize image to INPUT_IMAGE_RESHAPE
+        image = cv2.resize(image, self.INPUT_IMAGE_RESHAPE)
+
+        # Read the mask in grayscale mode
+        mask = cv2.imread(self.masks_fps[i], 0)
+
+        # Update the mask: Set FOREGROUND_CLASS to 1 and the rest to 0
+        mask_remap = np.where(mask == self.FOREGROUND_CLASS, 1, 0).astype(np.uint8)
+
+        # resize mask to INPUT_IMAGE_RESHAPE
+        mask_remap = cv2.resize(mask_remap, self.INPUT_IMAGE_RESHAPE)
+
+        if self.augmentation:
+            sample = self.augmentation(image=image, mask=mask_remap)
+            image, mask_remap = sample["image"], sample["mask"]
+
+        # Convert to PyTorch tensors
+        if image.ndim == 2:
+            image = np.expand_dims(image, axis=-1)  # Add channel dimension if missing
+        image = (
+            torch.tensor(image).float().permute(2, 0, 1) / 255.0
+        )  # HWC -> CHW and normalize to [0, 1]
```

Comment:
Comment on top to avoid line break while applying formatting
```suggestion
        # HWC -> CHW and normalize to [0, 1]
        image = torch.tensor(image).float().permute(2, 0, 1) / 255.0
```

Reply:
Agreed and updated.